cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -219,9 +322,9 @@ def validate_operation(self, operation: cirq.Operation) -> None:
         Raises:
             ValueError: The operation isn't valid for this device.
         """
-        # TODO(#5050) uncomment once gateset logic is implemented
-        # if operation not in self._metadata.gateset:
-        #     raise ValueError(f'Operation {operation} is not a supported gate')
+
+        if operation not in self._metadata.gateset:
+            raise ValueError(f'Operation {operation} is not a supported gate')
```

Comment:
`Operation` can be a circuit operation, in which case it won't have an underlying gate `op.gate`.  
 
```suggestion
            raise ValueError(f'Operation {operation} is not supported.')
```